dev/core/src/com/google/gwt/dev/jjs/impl/DeadCodeElimination.java

```diff
@@ -1206,17 +1221,21 @@ private boolean tryReduceSwitchWithConstantInput(JSwitchStatement s, Context ctx
 
       // Find the matching case
       JCaseStatement matchingCase = null;
-      for (JStatement subStatement : s.getBody().getStatements()) {
+      statements: for (JStatement subStatement : s.getBody().getStatements()) {
         if (subStatement instanceof JCaseStatement) {
           JCaseStatement caseStatement = (JCaseStatement) subStatement;
-          if (caseStatement.getExpr() == null) {
+          if (caseStatement.isDefault()) {
             // speculatively put the default case into the matching case
             matchingCase = caseStatement;
-          } else if (caseStatement.getExpr() instanceof JValueLiteral) {
-            JValueLiteral caseValue = (JValueLiteral) caseStatement.getExpr();
-            if (caseValue.getValueObj().equals(targetValue.getValueObj())) {
-              matchingCase = caseStatement;
-              break;
+          } else {
+            for (JExpression expr : caseStatement.getExprs()) {
+              if (expr instanceof JValueLiteral) {
+                JValueLiteral caseValue = (JValueLiteral) expr;
+                if (caseValue.getValueObj().equals(targetValue.getValueObj())) {
+                  matchingCase = caseStatement;
+                  break statements;
```

Comment:
See about rewriting to avoid the labeled break, maybe a stream/optional operation?